### DISCUSSION TITLE
Add support for Metal in macOS images

### DISCUSSION THREAD

**Discussion by Aaron1011** [community/community#160669] [Actions]
As described in https://github.com/actions/runner-images/issues/1779, Metal is not currently available on Github Actions macOS images.

Having Metal available (even if it's some kind of slow software implementation) would be a huge help for the Ruffle project (https://github.com/ruffle-rs/ruffle). As part of our Github Actions workflow, we render images on Linux and Windows runners, and compare them to expected images checked into the repository. However, we cannot run these image-based tests on macOS runners, resulting in a gap in our test coverage.

This would benefit any project that needs to perform GPU-based rendering on computation on macOS.

**Comment by miketimofeev** [comment]
@Aaron1011 There is no ETA for now but it's on our radar

**Reply by wcandillon** [reply]
@miketimofeev thank you for your reply. We are also deeply interested into this feature for https://github.com/shopify/react-native-skia. Are they any workarounds we could use in the meantime? Self-hosted runners?

**Reply by Skycoder42** [reply]
This is also relevant for flutter projects, as recent flutter based applications cannot run without vulkan support.

See https://github.com/flutter/flutter/issues/118469

**Comment by lslzl3000** [comment]
@miketimofeev besides metal on Macos, does DX11/12 supported in Windows or Vulkan  in Ubuntu runners?
As WebGPU is coming soon, I think Github runners should support those graphic drivers, It is hard to use Github actions to test projects rely on WebGPU rightnow, which basically use dx/vulkan/metal on each os

**Reply by Aaron1011** [reply]
@lslzl3000 On a Windows runner, you can use the WARP software rasterizer (https://learn.microsoft.com/en-us/windows/win32/direct3darticles/directx-warp). On Ubuntu, you can install `mesa-vulkan-drivers` to get the 'lavapipe' Vulkan software rasterizer.

**Reply by lslzl3000** [reply]
Thanks, is there a simple workflow demo to learn how to config a software rasterizer?

**Comment by damies13** [comment]
I am building an Open source app using the Kivy framework, an it too is expecting an OpenGL 2.0+ back end.

- For Widows i found a workaround by installing [mesa](https://github.com/pal1000/mesa-dist-win) drivers into windows
- For MacoS i haven't found an answer or workaround yet
- Haven't looked for a solution for Linux yet

Hope this is helpful to someone at least
